File: probs/1557.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        int ins[n];
        for (int idx = 0; idx < n; idx++) {
            ins[idx] = 0;
        }

        for (int idx = 0; idx < edges.size(); idx++) {
            int to = edges[idx][1];

            ins[to]++;
        }        

        vector<int> result;

        for (int idx = 0; idx < n; idx++) {
            if (ins[idx] == 0) {
                result.push_back(idx);
            }
        }

        return result;
    }
};

File: probs/1557.rs
impl Solution {
    pub fn find_smallest_set_of_vertices(
        n: i32,
        edges: Vec<Vec<i32>>
    ) -> Vec<i32> {
        let mut ins = vec![0; n as usize];

        for pair in edges.iter() {
            let to = pair[1] as usize;

            ins[to] += 1;
        }

        let mut result = Vec::new();

        for (idx, val) in ins.iter().enumerate() {
            if *val == 0 {
                result.push(idx as i32);
            }
        }

        result
    }
}